File: Algorithms/Warmup/Time Conversion.py
import os

#
# Complete the 'timeConversion' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def timeConversion(s):
    # Write your code here
    time_parts = s.split(':')
    if 'PM' in s:
        if time_parts[0] == '12':
            time_parts[0] = '12'
        else:
            time_parts[0] = str(int(time_parts[0]) + 12)
    else:
        if time_parts[0] == '12':
            time_parts[0] = '00'
    return ":".join(time_parts[:-1]) + ":" + time_parts[-1][:-2]


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')
    s = input()
    result = timeConversion(s)
    fptr.write(result + '\n')
    fptr.close()

File: Algorithms/Warmup/Time Conversion.php
<?php

/*
 * Complete the 'timeConversion' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

function timeConversion($s) {
    # Write your code here
    $period = substr($s, -2);
    $time = explode(':', $s);
    $hour = $time[0];
    $minute = $time[1];
    $second = substr($time[2], 0, 2);

    if ($period == 'PM' && $hour != '12')
        $hour = (int)$hour + 12;
    elseif ($period == 'AM' && $hour == '12')
        $hour = '00';

    return implode(':', [$hour, $minute, $second]);
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");
$s = rtrim(fgets(STDIN), "\r\n");
$result = timeConversion($s);
fwrite($fptr, $result . "\n");
fclose($fptr);
